src/dlgProfilePreferences.cpp

```diff
@@ -1763,8 +1780,26 @@ void dlgProfilePreferences::slot_save_and_exit()
 
     mudlet::self()->setToolBarIconSize(MainIconSize->value());
     mudlet::self()->setEditorTreeWidgetIconSize(TEFolderIconSize->value());
-    mudlet::self()->setMenuBarVisible(showMenuBar->isChecked());
-    mudlet::self()->setToolBarVisible(showToolbar->isChecked());
+    switch (comboBox_menuBarVisibility->currentIndex()) {
+    case 0:
+        mudlet::self()->setMenuBarVisibility(mudlet::visibleNever);
+        break;
+    case 1:
+        mudlet::self()->setMenuBarVisibility(mudlet::visibleOnlyWithoutLoadedProfile);
```

Comment:
What is this idea behind this new option?

Reply:
If you want to run a single profile without the main toolbar or the menu showing (say you have a fancy MUD specific GUI that you want to devote the absolute maximum screen space to) this means that you can have them still present on start up (needed so you can choose which profile to load) and _then_ they will hide themselves once that profile starts up.

Reply:
The settings for the two controls concerned are inter-linked so you can not set BOTH controls to be "Never shown" - one must be at least "Visible before a profile is loaded" so we try to prevent the hiding both sets of controls on _start-up_ - but allow it once a profile is loaded.

Reply:
> this means that you can have them still present on start up (needed so you can choose which profile to load)

The profile window opens automatically now, so there's no need to press `Connect` anymore on Mudlet start.

I don't see this option being all that useful still...

Reply:
... but if you hit *Cancel* on the "Connection Profile" dialog? 😜 

One thing I was considering for the future was if the Mudlet application does not close down with the last profile - which would be a potential side effect if we got a "Close" button/menu item (back ?) into the application - then you would need at least one of the menu bar or toolbar to come back - and there would not be any `TConsole`s around with a context menu with the emergency restoration actions on.

The reason I mention the close profile action is that I spotted there is a `(void) mudlet::slot_close_profile()` connected to `(QAction*) mactionCloseProfile` but the QAction is never placed anywhere! As it is, I do feel that there should be both a *Close* and a *Quit* menu option on the first menubar submenu; these are sort of expected I feel - and if you are running Mudlet in full-screen mode you will not have a title bar to click the "close application" button on there.  Hell, until I changed my Desktop styling options on the FreeBSD KDE desktop I am currently using, **none** of the windows on my desktop had any buttons - which was a little awkward but not unusable, as everything else had a menubar...

Of course even though I have an icon for the "Close profile" action already created, the toolbar is getting a little full and using a big size (3 or 4) on my 1650x1050 monitor does fill the screen width.  So we may want to include a user menu configuration option in the future so that they can choose which buttons they want and in what order - and permit the inclusion of one or two currently menu only options {_Disconnect_ and _Package Exporter_ for instance}.

Reply:
> ... but if you hit Cancel on the "Connection Profile" dialog?

Change the name of two options in settings files, keep code around for 4 options forever, break things for people using different Mudlet versions across computers with the same profiles, just so somebody who hits cancel doesn't have to spend 5 seconds to re-open Mudlet?

I don't think the code-benefit analysis is in favour here 🤔 

Let's remove the 3rd option, keep two options in settings files, and keep the rest of the code improvements you did.

Reply:
I have pointed out that we can drop the backwards compatibility with Mudlet 4.0.x (or 4.1.x) for the old settings. Basically though I am thinking ahead because I do want to be able to handle gracefully the option, which we do not currently support, but which I think is not unreasonable, in that closing the last active profile _need_ not close the whole application.

For instance, say you are playing a MUD and running a background/non-profile related IRC session to see which _side_ you want to long on before starting up. WoTMUD, for instance, requires a five/fifteen minutes break before you change character on the same / other two sides so you'd want to keep Mudlet running and chatting on IRC before connecting with a different profile.  At present that is not do-able but would be feasible - provided there were controls that, at least, popped up whenever there is not a profile loaded.  The current auto-showing of the "Connection Preferences" dialog _is_ useful but does not fit _every_ situation.

Reply:
... also, of course, when debugging it is useful to be able to close and reopen a profile without having to shut down the whole application, and at present that requires having a second one loaded... 😉

Reply:
> we can drop the backwards compatibility with Mudlet 4.0.x (or 4.1.x) for the old settings. 

Won't be dropping backwards compatibility in 4.x for anything... that's just stuff that annoys users. Someone will have to suck it up and it'll be us, for as long as we can.

Agree with you on keeping Mudlet open though when no profiles are still open, that is useful.

Reply:
What I was suggesting is that in 4.1(?) we no longer write out the old settings - so that anyone use a later version will not have the old, unused ones littering the QSettings pulled from the `mudlet.conf` - and _then_ toolbar/menu visibility settings for mudlet 3.7.1 and before will still be changeable for those old versions should they be used but they may in use diverge from those of later versions; the **only** people who might have issues with this is, presumably a very small number who _flip-flop_ between using a 3.7.1 or earlier version and a 4.x version on the same machine, so that their pre-3.7.1 version might have a different setting compared to the 4.x version and even then a visit to the preferences for the version concerned will fix the setting for that one...